lib/metasploit/framework/login_scanner/mysql.rb

```diff
@@ -64,8 +68,15 @@ def attempt_login(credential)
             })
           end
 
-          unless result_options[:status]
+          if mysql_conn
             result_options[:status] = Metasploit::Model::Login::Status::SUCCESSFUL
+
+            # This module no long owns the socket, return it as proof so the calling context can perform additional operations
+            # Additionally assign values to nil to avoid closing the socket etc automatically
+            if use_client_as_proof
+              result_options[:proof] = mysql_conn
+              mysql_conn = nil
+            end
           end
```

Comment:
This is leaking file descriptors; similar to the mssql PR https://github.com/rapid7/metasploit-framework/pull/18696#issuecomment-1918220512

lsof:

```
ruby    19006 user   14u  IPv4 0x3f021c11117b81af      0t0                 TCP localhost:51805->localhost:pyrrho (ESTABLISHED)
ruby    19006 user   15u  IPv4 0x3f021c11123a104f      0t0                 TCP 192.168.123.1:51859->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   16u  IPv4 0x3f021c111239fa0f      0t0                 TCP 192.168.123.1:51860->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   17u  IPv4 0x3f021c11117a81af      0t0                 TCP 192.168.123.1:51871->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   18u  IPv4 0x3f021c11117a6b6f      0t0                 TCP 192.168.123.1:51872->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   19u  IPv4 0x3f021c11117a552f      0t0                 TCP 192.168.123.1:51874->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   20u  IPv4 0x3f021c11117a4a0f      0t0                 TCP 192.168.123.1:51875->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   21u  IPv4 0x3f021c1111ad1e2f      0t0                 TCP 192.168.123.1:51853->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   22u  IPv4 0x3f021c1111ad130f      0t0                 TCP 192.168.123.1:51854->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   23u  IPv4 0x3f021c1111acf1af      0t0                 TCP 192.168.123.1:51856->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   24u  IPv4 0x3f021c1111acfccf      0t0                 TCP 192.168.123.1:51857->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   25u  IPv4 0x3f021c11123a47ef      0t0                 TCP 192.168.123.1:51862->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   26u  IPv4 0x3f021c111116d7ef      0t0                 TCP 192.168.123.1:51863->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   27u  IPv4 0x3f021c111116ab6f      0t0                 TCP 192.168.123.1:51865->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   28u  IPv4 0x3f021c111116cccf      0t0                 TCP 192.168.123.1:51866->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   29u  IPv4 0x3f021c11117a8ccf      0t0                 TCP 192.168.123.1:51868->192.168.123.1:mysql (ESTABLISHED)
ruby    19006 user   30u  IPv4 0x3f021c11117a768f      0t0                 TCP 192.168.123.1:51869->192.168.123.1:mysql (ESTABLISHED)
```

Can we use the same approach as the postgres client to ensure any open sockets are closed:

https://github.com/rapid7/metasploit-framework/blob/15d22d2807a0ef9fe10599728cc92166195e9744/lib/metasploit/framework/login_scanner/postgres.rb#L76-L89

Reply:
Got this updated and working. Good catch! 👍